src/Voron/Data/CompactTrees/CompactTree.cs

```diff
@@ -1958,7 +1958,11 @@ int GetEncodedEntryUnlikely(byte* p, out int length)
 
             key = new Span<byte>(entryPos + lenOfKeyLen, keyLen);
             entryPos += keyLen + lenOfKeyLen;
-            value = ZigZagEncoding.Decode<long>(entryPos, out var valLen);
+
+            value = ZigZagEncoding.DecodeCompact<long>(entryPos, out var valLen, out var success);
```

Comment:
Shouldn't this be the same behavior? What is the difference?

Reply:
I was not checking for error explicitly.

Reply:
How would you _get_ the error, however?

Reply:
Different error, now we throw the same exception in both cases. Also notice that we are using the `Compact` version now (shorter codegen).